code/__HELPERS/objectives.dm

```diff
@@ -0,0 +1,37 @@
+/proc/check_ship_objectives()
+	var/go_home = FALSE
+	if(world.time > 54000 || !get_ship_objective_types_len())
+		go_home = TRUE
+	for(var/datum/objective/O in get_ship_objectives())
+		O.check_completion()
+		if(!go_home && (O.failed || O.completed))
```

Comment:
Selling any item with atleast one completed objective will generate a new objective. Maybe check to see if an objective isn't finished before checking completion -> new objective.

Reply:
okay, so you want it so that once you've completed all objectives, you then start to get more objectives?

Reply:
like you;ve done aa a a a